src/parser/warrior/arms/modules/core/Slam.js

```diff
@@ -0,0 +1,53 @@
+import React from 'react';
+
+import { formatPercentage } from 'common/format';
+import SpellLink from 'common/SpellLink';
+import SPELLS from 'common/SPELLS';
+import Analyzer, { SELECTED_PLAYER } from 'parser/core/Analyzer';
+import Events from 'parser/core/Events';
+import SpellUsable from 'parser/warrior/arms/modules/features/SpellUsable';
+
+class Slam extends Analyzer {
+  static dependencies = {
+    spellUsable: SpellUsable,
+  };
+
+  badCast = 0;
+  totalCast = 0;
+
+  constructor(...args) {
+    super(...args);
+    this.addEventListener(Events.cast.by(SELECTED_PLAYER).spell(SPELLS.SLAM), this._onSlamCast);
+  }
+
+  _onSlamCast() {
```

Comment:
This also receives `event` as a parameter (see LOD module), but because it's not used here, it's probably fine this way.
https://github.com/WoWAnalyzer/WoWAnalyzer/blob/d5a23314cfa621d63f9df8d6cce9cc5b262125e1/src/parser/paladin/holy/modules/core/LightOfDawn.js#L9

Reply:
Oh I didn't know that, I guess I could change the old way of listening to events on other files at some point

Reply:
Of course, it's *event* listener after all 😄 I guess it will become the preferred way of handling events, but both ways are supported now (not combined though).

Reply:
Yeah that's make sense 😅